### DISCUSSION TITLE
哔哩哔哩用一段时间后内存占用会变大

### DISCUSSION THREAD

**Discussion by jhs114514** [ywmoyue/biliuwp-lite#369] [bug]
### Bug 描述

哔哩哔哩用一段时间后内存占用会变大，第一张图是刚打开软件时的内存占用，第二张图是使用软件一段时间后的内存占用

### 操作步骤

```plain text
1.打开软件
2.不断点击新视频然后关闭
```

### 期望结果

希望软件在长时间使用后内存占用不会增加太多

### 截图

![哔哩1](https://github.com/ywmoyue/biliuwp-lite/assets/149226254/1c0faaaf-43d1-4978-883f-f46d183c8388)
![哔哩2](https://github.com/ywmoyue/biliuwp-lite/assets/149226254/93a199a0-edbe-4a2e-ac79-04dacfea49ec)

### 应用来源

Github/其他Github镜像站

### 代码贡献

我更擅长发现问题

**Comment by GD-Slime** [comment]
看了一下内存分析, 似乎在这样一次打开视频的流程中: 
![image](https://github.com/ywmoyue/biliuwp-lite/assets/82302542/f03fe810-0d6d-4f9b-80a1-4e3d508d2931)
两次内存快照差值增长的主要来源是加载的评论区:
![image](https://github.com/ywmoyue/biliuwp-lite/assets/82302542/2b29e02b-751a-418d-833f-81a39ae31761)
是否是这个原因导致的打开多个视频后，评论区的数据不能完全卸载导致内存增长呢？@ywmoyue

**Reply by ywmoyue** [reply]
有可能，我检查下注入的视图模型的生命周期

**Reply by ywmoyue** [reply]
是这个问题https://github.com/microsoft/microsoft-ui-xaml/issues/3303

试了下从改了个x:bind为binding，对应的内存就没泄露了

**Comment by GD-Slime** [comment]
此问题仍然存在， 大约每打开-关闭一个视频界面就会多出20 - 50mb内存…… 大约只需要看20个视频就能占满1g.
我尝试将例如评论区和EmotePackage等元素在关闭页面后手动删除，但是也收效甚微。 
有点困惑，这个问题到底是什么东西引起的？

**Reply by ywmoyue** [reply]
这块我看了是非托管内存的泄漏，可能是播放器调用ffmpeg引起的，播放器目前有点复杂，我重构完播放器之后再看看

**Comment by GD-Slime** [comment]
https://github.com/ffmpeginteropx/FFmpegInteropX/releases/tag/v1.0.1 
在1.0.1版本的FFmpegInteropX中修复了一个内存泄露的bug. 
现在项目使用的版本为1.0.0. 我想可能与这里的内存泄露有关？ 希望对问题有帮助 @ywmoyue

**Reply by ywmoyue** [reply]
@GD-Slime 非常感谢，我试一试

**Reply by ywmoyue** [reply]
简单测了下似乎没有效果，后面再看看吧
